scripts/mellanox_buffer_migrator.py

```diff
@@ -8,263 +78,369 @@
+            "buffer_profiles": {
+                "default": {"ingress_lossless_profile": {"dynamic_th": "7", "pool": "[BUFFER_POOL|ingress_lossless_pool]", "size": "0"},
+                               "ingress_lossy_profile": {"dynamic_th": "3", "pool": "[BUFFER_POOL|ingress_lossy_pool]", "size": "0"},
+                               "egress_lossless_profile": {"dynamic_th": "7", "pool": "[BUFFER_POOL|egress_lossless_pool]", "size": "0"},
+                               "egress_lossy_profile": {"dynamic_th": "7", "pool": "[BUFFER_POOL|egress_lossy_pool]", "size": "9216"},
+                               "q_lossy_profile": {"dynamic_th": "3", "pool": "[BUFFER_POOL|egress_lossy_pool]", "size": "0"}},
+                "singlepool": {"ingress_lossless_profile": {"dynamic_th": "7", "pool": "[BUFFER_POOL|ingress_lossless_pool]", "size": "0"},
+                               "ingress_lossy_profile": {"dynamic_th": "3", "pool": "[BUFFER_POOL|ingress_lossless_pool]", "size": "0"},
+                               "egress_lossless_profile": {"dynamic_th": "7", "pool": "[BUFFER_POOL|egress_lossless_pool]", "size": "0"},
+                               "egress_lossy_profile": {"dynamic_th": "7", "pool": "[BUFFER_POOL|egress_lossy_pool]", "size": "9216"},
+                               "q_lossy_profile": {"dynamic_th": "3", "pool": "[BUFFER_POOL|egress_lossy_pool]", "size": "0"}}
+            }
+        },
+        "version_1_0_5": {
+            # version 1.0.5 is introduced for shared headroom pools
+            "pool_convert_map": {
+                "spc1_t0_pool_sku_map": {"Mellanox-SN2700-C28D8": "spc1_2700_t0_pool_shp",
+                                         "Mellanox-SN2700-D48C8": "spc1_2700-d48c8_t0_pool_shp",
+                                         "Mellanox-SN2700": "spc1_2700_t0_pool_shp"},
+                "spc1_t1_pool_sku_map": {"Mellanox-SN2700-C28D8": "spc1_2700_t1_pool_shp",
+                                         "Mellanox-SN2700-D48C8": "spc1_2700-d48c8_t1_pool_shp",
+                                         "Mellanox-SN2700": "spc1_2700_t1_pool_shp"},
+                "spc2_3800_t0_pool_sku_map": {"Mellanox-SN3800-C64": "spc2_3800-c64_t0_pool_shp",
+                                              "Mellanox-SN3800-D112C8": "spc2_3800-d112c8_t0_pool_shp",
+                                              "Mellanox-SN3800-D24C52": "spc2_3800-d24c52_t0_pool_shp",
+                                              "Mellanox-SN3800-D28C50": "spc2_3800-d28c50_t0_pool_shp"},
+                "spc2_3800_t1_pool_sku_map": {"Mellanox-SN3800-C64": "spc2_3800-c64_t1_pool_shp",
+                                              "Mellanox-SN3800-D112C8": "spc2_3800-d112c8_t1_pool_shp",
+                                              "Mellanox-SN3800-D24C52": "spc2_3800-d24c52_t1_pool_shp",
+                                              "Mellanox-SN3800-D28C50": "spc2_3800-d28c50_t1_pool_shp"}
+            },
```

Comment:
Are `spc2_3800_t0_pool_sku_map` and `spc2_3800_t1_pool_sku_map` still necessary?
There aren't any 3800 buffer pool flavors defined in any version earlier than `1.0.5`.